### ISSUE TITLE
Possible issue with GSL version and CONCOCT?

### ISSUE THREAD

**Issue by jarrodscott** [merenlab/anvio#827]
Howdy. 

Not sure if this is a widespread problem but I did want to post this in case anyone else has a similar issue.
Relevant info:

`Anvi'o version ...............................: 4-master, "rosalind"`
`Profile DB version ...........................: 26`
`Contigs DB version ...........................: 12`
`Pan DB version ...............................: 9`
`Genome data storage version ..................: 6`
`Auxiliary data storage version ...............: 2`

`Linux login-30-2.local 2.6.32-696.23.1.el6.centos.plus.x86_64 #1 SMP Wed Mar 14 11:51:06 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`

Ok. I have been banging my head against a wall trying to figure out why anvi'o would stall out when I ran:

`anvi-cluster-with-concoct -c CONTIGS.db -p PROFILE.db --num-clusters 200 -C CONCOCT`

I submit my jobs with a script to a server and get a log file back--the file would just end like this and the job would terminate without further explanation:
`CONCOCT INIT .................................: Complete for 50840 splits`
`CONCOCT VGBMM ................................: Requesting 200 clusters`                                                                                
`[0;30m[46m`
`[09 May 18 12:37:05 VBGMM] Clustering ...   `

Long story short I tried the `master` build through `virtualenv` and a `conda` install--I tried a whole bunch of different things--but the result was always the same. Since this command worked for me in the past, I started looking at old `anvio` versions on `conda` (https://anaconda.org/bioconda/anvio/files) and I noticed that the only difference between the last two uploads (`py36_1` and `py36_2` ) is the version of `GSL` (`v.1.16` vs `v2.2.1'). 

Did a fresh install of the `master` but this time with `v.1.16` installed in my `conda` package. And now it seems to work just fine .

Could this be an issue with something else entirely  (e.g., gcc version, I use `gcc 4.9.2`). My `self tests` never failed, including the full test.

**Comment by ozcan** [comment]
Hey @jarrodscott,
Thanks for the heads up. Not on local installations but when I install on our cluster virtualenv, I change gsl paths in `setup.py` due to not standart library location. But I strongly believe that compilation was correct each time but computation cluster was killing your process due to some kind of hard or soft limitation like child count of a process /thread count / memory usage. Because otherwise it should have failed during compilation step. Different compilers can link different libraries that causing this termination. 
I think you can check your clusters logs to find out the real reason for termination. But I don't think there are much we can do about this on our end.
Best,

**Comment by jarrodscott** [comment]
THanks Ozcan. I figured this was not an issue with anvio but thought this
may be useful to someone :) perhaps I should have posted this to the
forum...

Do you also change the setup.py script in the extensions/concoct directory
or just the main setup.py script?

**Comment by ozcan** [comment]
Hey Jarrod,
I only change `ext_modules` in main `setup.py`. I have very old code snippet here https://gist.github.com/ozcan/6ece647ef8514e0ee1ff
Basically, I extend `include_dirs` and add `extra_objects` keyword argument that contains relevant directory paths.
Best,
